File: sync/src/main/scala/com/fan/mysql/event/AbstractEvent.java
package com.fan.mysql.event;

public abstract class AbstractEvent implements BinlogEvent {

    protected EventHeader header;
    protected byte[] originData;

    public AbstractEvent(EventHeader header) {
        this.header = header;
    }

    public EventHeader getEventHeader() {
        return header;
    }

    public byte[] getOriginData() {
        return originData;
    }

    public void setOriginData(byte[] originData) {
        this.originData = originData;
    }

}

File: async/src/main/scala/com/fan/mysql/async/binlog/event/AbstractEvent.scala
package com.fan.mysql.async.binlog.event

abstract class AbstractEvent(header: EventHeader) extends BinlogEvent {

  protected var originData: Array[Byte] = _

  override def getEventHeader: EventHeader = header

  override def getOriginData: Array[Byte] = originData

  override def setOriginData(originData: Array[Byte]): Unit = {
    this.originData = originData
  }

}
